### PR TITLE
[1.8.8] Fuel handler improvement - allow to make fuel value smaller or 0

### PR DESCRIPTION
**Pull request by liach**
1.8.8 version of #2186, might not hurt compatibility.
Generally works the same, improved patches, test and javadoc, and I do not think that lex and cpw have time currently to make a better one.

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java [hunk 1]
```diff
@@ -297,14 +299,19 @@ public static void registerFuelHandler(IFuelHandler handler)
...
     public static int getFuelValue(ItemStack itemStack)
     {
-        int fuelValue = 0;
-        for (IFuelHandler handler : fuelHandlers)
+        ListIterator<IFuelHandler> li = fuelHandlers.listIterator(fuelHandlers.size());
+        while (li.hasPrevious())
         {
-            fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack));
+            int fuelValue = li.previous().getBurnTime(itemStack);
+            if (fuelValue > 0)
```
Comment (Spartan322) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: Wouldn't `>=` be more consistent

Comment (liach) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: 0 means that it is not your item. see the improved javadoc [here](https://github.com/liachmodded/MinecraftForge/blob/fuel-1.8.8/src/main/java/net/minecraftforge/fml/common/IFuelHandler.java#L19-L31).

Comment (Spartan322) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: nevermind, missed the old method.

**Comment by Spartan322** [comment]
Only thing left I have to say is, isn't this is technically adding functionality, so shouldn't the PR name be changed to something slightly more descriptive (I got confused of the purpose for a minute both by that comment and PR name)

**Comment by liach** [comment]
So I targeted this pr to 1.8.8 beta version

**Comment by cpw** [comment]
A thorough rewrite is in progress here. I'd rather not muddy the water by supporting the hurried patch to 1.7.10 that was agreed would never see life in 1.8.+

**Comment by liach** [comment]
Is 97ef6a5d1fb7f9abd4052ab449b089533a2c7938 the rewrite?

**Comment by liach** [comment]
@cpw Could you fix this bug up? I may migrate this pr to 1.9.

**Comment by liach** [comment]
Would you merge a pr of this for 1.9? If yes I'll open one with already done code.

**Pull request closed without merging**